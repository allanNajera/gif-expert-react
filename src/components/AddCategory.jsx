import { useState } from 'react';
import {PropTypes} from 'prop-types';

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    /*Lo que sucede aquí es que el onChange sucede cada vez que se modifica 
    y es la forma que se hace para poder escribirlo y que se note en el input
    */
    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault(); //esto es para evitar que cuando le damos enter después de escribir refreque la pantalla
        if(inputValue.trim().length <= 1) return; //esto para no hacer nada si solo se está agregando algo con una letra o vacio
        //setCategories(categories => [...categories, inputValue]);
        onNewCategory(inputValue.trim());
        setInputValue('');
    }
    
    return (
        <form onSubmit={ onSubmit} aria-label="form">
            <input
                type='text'
                placeholder="Buscar gifs"
                value={inputValue}
                //onChange={ (event) => onInputChange(event) } Se puede hacer así o así:
                onChange={onInputChange}
                /*funciona igual porque siempre envia el event aunque no esté explicito,
                solo hay que asegurarnos de recibirlo y utilizarlo
                */
            />
        </form>

    )


}

AddCategory.propTypes = {
    onNewCategory : PropTypes.func.isRequired,
}
