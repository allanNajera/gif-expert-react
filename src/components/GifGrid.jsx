import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import {PropTypes} from 'prop-types';

export const GifGrid = ({ category }) => {
    const {images, isLoading} = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {isLoading && (<h2>Cargando...</h2>) /*Esto se hizo para mostrar el cargando cuando las imagenes a√∫n no se han cargado */}
            <div className="card-grid">
                {
                    images.map( (image) => (
                        <GifItem key={image.id} {...image} />
                    ))
                }
            </div>
        </>
    )
}


GifGrid.propTypes = {
    category : PropTypes.string.isRequired,
}