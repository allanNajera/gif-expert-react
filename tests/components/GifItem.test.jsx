
import { render,screen } from "@testing-library/react";

import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem/>', () => {
    const title = 'Saitama';
    const url = 'https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=27ddde81n6u4njvby9195crwrqupdf20lxtbhfl9u4rd5bgx&rid=giphy.gif&ct=g';

    test('debe hacer match con el snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el url y el alt indicado', () => { 
        render(<GifItem title={title} url={url}/>);
        //screen.debug();//Esto para conocer que tiene el screen
        //expect(screen.getByRole('img').src).toBe(url);
        //expect(screen.getByRole('img').alt).toBe(title);
        //En vez de hacer esas dos aserciones, hacemos:
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('debe de mostrar el titulo en el componente ', () => { 
        
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    });
});