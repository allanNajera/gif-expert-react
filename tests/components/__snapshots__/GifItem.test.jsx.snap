// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=27ddde81n6u4njvby9195crwrqupdf20lxtbhfl9u4rd5bgx&rid=giphy.gif&ct=g"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
